addons/binding/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/handler/OneWireGPIOHandler.java

```diff
@@ -0,0 +1,127 @@
+ */
+package org.openhab.binding.onewiregpio.handler;
+
+import static org.openhab.binding.onewiregpio.OneWireGPIOBindingConstants.*;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.onewiregpio.internal.OneWireGPIOSensorReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OneWireGPIOHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Anatol Ogorek - Initial contribution
+ */
+public class OneWireGPIOHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(OneWireGPIOHandler.class);
+
+    private String gpioBusFile;
+    private Integer refreshTime;
+
+    public OneWireGPIOHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(TEMPERATURE)) {
+            if (command instanceof RefreshType) {
+                publishSensorValue(channelUID);
+            } else {
+                logger.debug("Command {} is not supported for channel: {}. Supported command: REFRESH", command,
+                        channelUID.getId());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        checkConfiguration();
+        startAutomaticRefresh();
+        super.initialize();
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    /**
+     * This method checks if the provided configuration is valid.
+     * When invalid parameter is found, default value is assigned.
+     */
+    private void checkConfiguration() {
+        logger.debug("Start reading Feed Thing configuration.");
+        Configuration configuration = getConfig();
+
+        // It is not necessary to check if the URL is valid, this will be done in fetchFeedData() method
+        gpioBusFile = (String) configuration.get(GPIO_BUS_FILE);
+        if (gpioBusFile == null) {
+            gpioBusFile = DEFAULT_GPIO_BUS_FILE;
+            logger.warn("GPIO_BUS_FILE not set. Falling back to default value: {}. {}", DEFAULT_GPIO_BUS_FILE);
+        }
+
+        try {
+            refreshTime = (Integer) configuration.get(REFRESH_TIME);
+            if (refreshTime.intValue() <= 0) {
+                throw new IllegalArgumentException("Refresh time must be positive number!");
```

Comment:
don't throw an exception; set the status to OFFLINE-configuration-error instead.